### ISSUE TITLE
MHV on VA Landing Page links section analytics

### ISSUE THREAD

**Issue by LynneJohnson224** [department-of-veterans-affairs/va.gov-team#66723]
## Issue Description
As a team member supporting MHV on VA.gov, I want to be able to easily see click volume of individual links in the Resources and Support section ("My VA Health Benefits" and "More Resources and Support" sections/columns) on the MHV on VA.gov landing page so that I can be better informed when making decisions as to what landing page links to keep and which ones to remove and replace.

PR: https://github.com/department-of-veterans-affairs/vets-website/pull/26781

---
## Tasks 
- [ ] Look into segment of users who do not have VA health care but are registered at a facility? Is it understood by the team? @wesrowe 
- [x] Check design system and analytics documentation to learn if alert component out of the box tracks clicks in GA @fmccaf1 

## Acceptance Criteria
- Resources and Support links are grouped in such a way that allows us to add new links to that section, and they become part of the analytics for Resources and support clicks
- Both individual links and aggregate click volume are easily accessed
- The user can see how click volume is changing over time for both aggregate Resources and Support link clicks and individual links within that section

## Resources/Notes

- [Platform Analytics & Insights docs](https://depo-platform-documentation.scrollhelp.site/analytics-monitoring/getting-started-with-platform-analytics-insights). Documents how Google Analytics and Domo are set up for va.gov
- [Datadog Monitoring](https://depo-platform-documentation.scrollhelp.site/developer-docs/monitor-applications-and-infrastructure-with-datad). Datadog has realtime user monitoring (RUM), but overall Datadog is oriented toward "is this working/performing as expected" uses.

## Testing Instructions:
- _Google Analytics is not available in staging. It is possible to test the branch locally, or when the merge has been completed._ 

1. Install the **Adswerve** chrome extension and restart the browser
2. Go to va.gov/my-health/
3. Scroll down to the section labeled **My VA health Benefits**
4. Open up the **Console** by right clicking anywhere on the page and click on **Inspect**, this will bring up a screen with your browser.
5. Make sure **Console** is the open tab
6. Click on the link **Current Veteran copay rates** in the section labeled **My VA health Benefits** 
7. In the **Console**, verify that you see any information which resembles the following screenshot. This is **Adswerve** showing the information being recored for this link. <img src="https://assets.codepen.io/282917/full_screen_console_open.png" alt="" />
8. In the **Console**, verify that you see the following information:
        event: "nav-linkslist",
	links-list-title: "Current Veteran copay rates",
	links-list-group: "My VA health benefits"
9. Try clicking additional links and verify that information is appearing in the Console that contains 
**a: event 
  b: links-list-title 
  c: links-list-group:** properties. 
<img src="https://assets.codepen.io/282917/console_link_example.png" alt="" />

**Comment by wesrowe** [comment]
@MikeC-A6, based on my research into our current GA situation, we need to implement an analytics scheme for every link on the page, not just the ones at the bottom. We are nearly blind right now to how the landing page is being used, as events only fire on a handful of our links. I believe we need to bring the page up to parity with other auth exp pages, e.g. Profile and My VA. (I met with AJ from My VA to validate my observations.)

I figured we would edit this one to capture GA work for the whole page, we can discuss in planning.

**Comment by wesrowe** [comment]
Notes:

1) In planning we noted that we should look into whether the ACs can be met using Datadog (specifically RUM).

2) I just found Analytics team docs for [Navigation Events](https://depo-platform-documentation.scrollhelp.site/analytics-monitoring/navigation-events). Look in the table for link-list. I think this might essentially tell us how to implement GA events in front-end javascript in a way that will leverage GTM's existing wiring to make it through to the GA platform/reporting.

**Comment by radavis** [comment]
Utility function being used to report events to GA: https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/platform/monitoring/record-event.js

**Comment by fmccaf1** [comment]
## Notes from meeting w Wes and Florence on 10/25

Our goal today / now: Look at the page and ask ourselves, “what are the ways we’d want to measure this?”

### Things we’re curious about– sorted by whether the My VA implementation serves the curiosity:
#### My VA analytics strategy solves:
- Differentiated clicks among the various lists of links (e.g., Medications, My VA health benefits)
- Clicks on “Learn more about…” at the top of page

#### My VA analytics strategy doesn’t solve:
- (New?) segment of users who are not registered at a facility 
           - maybe add a GA event to the new alert box (if not present): Solved by adding an event to alert box (if not automatic)
           - AR: check design system docs about alerts: Florence?
- (New?) segment of users who do not have VA health care but are registered at a facility? Is it helpful to know percentage of users and what they click on? Includes people who visited ER, vaccination… 
          - Do they reach the landing page today? 
          - Do they have any specific experience on MHV National Portal that would enable segmentation?
          - AR: Wes look into this path; is it understood by the team?
- Should we bucket all gray box links and compare with all links in columns on bottom part of screen (My VA health benefits, More resources and support, and In the spotlight)?
- Can we get a 3rd parameter into the GA event: tool link vs. non-tool link?

### Hypotheses:
Users who are not registered to a facility will be more likely to click a resources/support link than users who can see the whole MHV landing page (tools included)

**Comment by wesrowe** [comment]
I forgot to document here how the MyVA events are named – 

Example: `Navigation - Link List - Health care - View your message`

How I think this is built, and how we would use it:
- `Navigation - Link List - ` come with the particular GTM/data dictionary entry we're using, I don't think we can affect these (confirm?)
- We have control over the last two terms – `- [Header above link] - [link text (can be abbreviated a little)]`

**Comment by wesrowe** [comment]
To @fmccaf1's point about [analytics that should come with the design system](https://design.va.gov/storybook/?path=/docs/components-va-alert--sign-in-or-tool-prompt), I would like help validating how those events work.

This one sounds promising: `va-component-did-load`. However, I went and found an info box that on the Find-a-form product ([staging link](https://staging.va.gov/find-forms/)), and don't see component loading in the AdSwerge GA debugger. 

I am probably misinterpreting the documentation. `va-component-did-load` doesn't sound like something you would typically see in GA, so maybe it's a javascript event or something? Anyway, since the team seems to be doing well on commitments I am hoping we can solve this puzzle.

**Comment by radavis** [comment]
If the goal is to track link clicks for our own purpose, we can specify action names on our links with a `data-dd-action-name` attribute: [Declare a name for click actions](https://docs.datadoghq.com/real_user_monitoring/browser/tracking_user_actions/#declare-a-name-for-click-actions)

**Comment by jzucadi** [comment]
va-component-did-load is a custom javascript event for accessibility. It’s tracking when the element loads in the DOM but this info isn’t relayed back to GA. The purpose is to call the `focusElement` on the alert as soon as it’s loaded, making it prominent for screen readers etc and non mouse users.

when a user clicks on a link within the va-alert component, this information is sent as a custom event to Google Analytics.

We can track user clicks in GA using the <va-link> component in the following way:  <va-link
            href=“/mental-health-link/“
            onClick={() => {
              recordEvent({
                event: `${GA_PREFIX}-Mental-Health-Services-link-clicked`,
              });
            }}
            text="Find VA facilities near you that offer Mental Health “Services
            data-testid="find-facilities-link"
          />

**Comment by wesrowe** [comment]
I documented intentions and potential naming conventions for GA and DD analytics. Lmk your thoughts, and certainly any challenges you encounter with trying to do this.

### Goals for implementation:
Labels/names should reflect 

- The link text
- The column header it’s under
- (if possible) the url destination – e.g., useful as tool links start to link to VAgov instead of National Portal

### Implementation details
#### GA:
Hypothesis: “link list” wiring that My VA uses will work for us
- Desired ‘event action’ in GA: Navigation - Link List - [column header] - [(link text)+(destination domain)]

Destination domain above would be (ideally) an indicator of either VAgov or National Portal, appended to the link text since we don’t have another field we can put it in. 
- vagov = va.gov
- mhvnp = national portal
- cerner = cerner (if we can figure that one out)
- unknwn = unknown

#### DD:
Using a ‘data’ attribute in markup, we can provide a custom “action” string for any link
- Proposed naming convention: [column header abbreviation]-[link text (shortened)]-[url]

**Comment by radavis** [comment]
See this use of `recordEvent` that includes the link title and link group: https://github.com/department-of-veterans-affairs/vets-website/blob/profile-contacts-configuration-update/src/applications/personalization/dashboard/components/benefit-payments/BenefitPayments.jsx#L34

**Comment by wesrowe** [comment]
Note to devs: review instance won't help us QA these GA events. I just loaded the [R.I. for another pr](http://374472cd5552dac83788a0a91517384d.review.vetsgov-internal/) and see no GA events at all. I don't even think that the GA javascript library is present, as I don't even see things like `dataLayer.push`.

**Comment by jzucadi** [comment]
Platform support has not determined why this review site for this PR does not work. The outstanding issue here is that the code is done, but we lack an actually staging/testing environment that allows us to test this. It is not possible to know what data is being collected at this time.  Suggest using a feature flag to control this

**Comment by wesrowe** [comment]
<img width="176" alt="image" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/4054752/c19f0bcb-3cdf-4f70-8439-943c7e1b2e47">

**Comment by LynneJohnson224** [comment]
@jzucadi want to make sure we're closing the loop on this ticket. Has this been merged? Are you looking for someone to review/QA it?

**Comment by jzucadi** [comment]
@LynneJohnson224 This merge is being held up by the main branch. The main branch received an update yesterday that caused several tests to fail, and the PR requires that my branch be synced with master before merging. There have been several commits to main branch since yesterday trying to fix this issue, but so far none have successful. I am monitoring the situation with the main branch closely and will update the ticket once a fix for main has been committed.

**Comment by wesrowe** [comment]
@jzucadi – good news and bad news... 

🎉Good news! 🎉 : your PR reached staging, and the event looks like it wired correctly in AdSwerve!
<img width="493" alt="image" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/4054752/b6954cdc-1502-4575-aadc-33a20224018f">

The bad news is, it doesn't include information that would allow us to know which link the event tracks or what section/group the link is in.

**Comment by wesrowe** [comment]
And please lmk if you think you could achieve the "destination domain" idea in the [comment a little ways above](https://github.com/department-of-veterans-affairs/va.gov-team/issues/66723#issuecomment-1789106226). I proposed this because it would help us segment clicks later between the National Portal and staying on VAgov. But it's a nice-to-have for this first implementation, so I would be fine with punting it to a future story.

**Comment by jzucadi** [comment]
I'm checking this out now, I will figure out a fix asap. I can work on the destination domain idea too, would like to get something done by tomorrow so we can test before monday

**Comment by jzucadi** [comment]
@wesrowe I have a PR for fixing the properties and adding the destination domain. I've tested it locally and it appears to work, will keep checking this and update the ticket with testing instructions

**Comment by wesrowe** [comment]
@LynneJohnson224, I just tested in Staging and still see the same as my [screenshot from 11/16](https://github.com/department-of-veterans-affairs/va.gov-team/issues/66723#issuecomment-1815318838) – i.e., the Link List event is firing, but without the necessary parameters. I'm going to reopen the story until we can demonstrate it working in Staging.

**Comment by jzucadi** [comment]
PR: https://github.com/department-of-veterans-affairs/vets-website/pull/26781

**Comment by wesrowe** [comment]
Notes from meeting with Jamie Stutt of Analytics team:
- the links at the bottom (R&S) are va-links, but they have `disable-analytics=true` props in html. was this because we disabled it in order to fire our own?
- the `destination-domain` parameter is problematic – 
<img width="380" alt="image" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/4054752/0b689ef5-2f50-4bfb-ba29-b5c2b4cfa659">
- put in a analytics request ticket for destination-domain
- there were events being collected until yesterday???!!! [GA report](https://analytics.google.com/analytics/web/#/report/content-event-pages/a50123418w177519031p176188361/_u.date00=20231201&_u.date01=20231201&explorer-table.plotKeys=%5B%5D&_r.drilldown=analytics.pagePath:www.va.gov~2Fmy-health~2F,analytics.eventCategory:Interactions/)

**Comment by wesrowe** [comment]
The notes above were more relevant actually to the segmentation story.

I tested the Resources/Support links included in this ticket, and they look good! 🎉

**Comment by jzucadi** [comment]
@wesrowe 
- yes, the idea was to disable it to fire our own
- I removed the destination domain function in a new PR: https://github.com/department-of-veterans-affairs/vets-website/pull/26970
- I submitted a ticket to the analytics team for the destination domain: https://github.com/department-of-veterans-affairs/va.gov-team/issues/71140
